extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -100,7 +100,11 @@ export abstract class AzureAuth implements vscode.Disposable {
 				}
 			}
 			Logger.error(ex);
-			return undefined;
+			return {
+				canceled: false
+			};
+		} finally {
+			loginComplete?.reject(undefined);
```

Comment:
Finally doesn't seem correct here - you only want to reject if an error occurs (and you already call reject if the exception is an AzureAuthError so there's a possibility of two rejects happening here)

Reply:
Promises don't take two rejects, if a resolve or reject happens it'll just stick to that one. Any other operation on it becomes a no-op to facilitate patterns like this

Reply:
Huh, TIL. Not sure I like that much - seems much clearer to just specifically resolve/reject when appropriate.

For example here - you're not going to be rejecting with the exception which doesn't seem like what we want here. If something is await the promise then it's going to get undefined for the error...

Reply:
Also - earlier you're rejecting with a string which we shouldn't be doing. All rejections should be ideally be Errors.

Reply:
The thing with finally is that it guarantees that the promise won't just sit there in a "pending" state if we forgot one of the conditionals. I'll change a few things to require the rejection to have an error at least per the typing, because right now its any.